src/modules/services/components/bulk/AssignServiceButton.tsx

```diff
@@ -0,0 +1,112 @@
+import CheckIcon from '@mui/icons-material/Check';
+import { LoadingButton } from '@mui/lab';
+import { ButtonProps } from '@mui/material';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { useBulkAssignMutations } from '../../hooks/useBulkAssignMutations';
+import ApolloErrorAlert from '@/modules/errors/components/ApolloErrorAlert';
+import { formatDateForGql } from '@/modules/hmis/hmisUtil';
+import { BulkServicesClientSearchQuery } from '@/types/gqlTypes';
+
+interface Props {
+  client: BulkServicesClientSearchQuery['clientSearch']['nodes'][0];
+  dateProvided: Date;
+  projectId: string;
+  serviceTypeId: string;
+  tableLoading?: boolean;
+  disabled?: boolean;
+  cocCode?: string;
+}
+
+const AssignServiceButton: React.FC<Props> = ({
+  client,
+  dateProvided,
+  projectId,
+  serviceTypeId,
+  tableLoading,
+  disabled,
+  cocCode,
+}) => {
+  const { bulkAssign, bulkRemove, loading, apolloError } =
+    useBulkAssignMutations();
+
+  // Locally disable the button while the table refetches, AFTER this specific mutation has completed
+  const [localDisabled, setLocalDisabled] = useState(false);
+
+  // When the entire table has finished re-fetching, remove the loading state for this button.
+  useEffect(() => {
+    if (!tableLoading) setLocalDisabled(false);
+  }, [tableLoading]);
```

Comment:
Some funkiness here. `useBulkAssignMutations` uses apollo `refetchQueries` to specify that the table query should always refetch after a mutation completes. During that refetch, we want to keep this button disabled.

Reply:
This seems sensible. Maybe it would be good to add what you've written here in the github PR comment as a comment in the code as well?

Reply:
> would it work to just disable the buttons when tableLoading is true? 

that would disabled all the buttons in the table. we only want to disable the buttons that just performed a mutation